algebra-core/src/curves/models/short_weierstrass_jacobian.rs

```diff
@@ -202,6 +202,22 @@ impl<P: Parameters> AffineCurve for GroupAffine<P> {
     fn mul_by_cofactor_inv(&self) -> Self {
         self.mul(P::COFACTOR_INV).into()
     }
+
+    fn from_xy_checked(x: Self::BaseField, y: Self::BaseField) -> Result<Self, ()> {
+        let infinity = x.is_zero() && y.is_zero();
```

Comment:
(0, 0) is a point on some curves (eg: those of the form (`y^2 = x^3 + ax`)), so setting it to the point at infinity isn't correct.

Reply:
oh right, didn't think of this :/